tests/model_registry/model_catalog/upgrade/test_catalog_upgrade.py

```diff
@@ -0,0 +1,361 @@
+        @retry(wait_timeout=120, sleep=10, exceptions_dict={AssertionError: [], ResourceNotFoundError: []})
+        def verify_custom_catalog_ingestion():
+            sources_response = execute_get_command(
+                url=f"{model_catalog_rest_url[0]}sources", headers=model_registry_rest_headers
+            )
+
+            # Assert custom catalog exists alongside default catalog
+            custom_catalog_found = any(source["id"] == TEST_DATA["catalog_id"] for source in sources_response["items"])
+            assert custom_catalog_found, (
+                f"Custom catalog {TEST_DATA['catalog_id']} not found in sources: {[s['id'] for s in sources_response['items']]}"  # noqa: E501
+            )
+
+            # Verify default catalog still exists
+            default_catalog_found = any(source["id"] == DEFAULT_CATALOG_ID for source in sources_response["items"])
+            assert default_catalog_found, (
+                f"Default catalog should still exist alongside custom catalog. Available sources: {[s['id'] for s in sources_response['items']]}"  # noqa: E501
+            )
+
+            return sources_response
+
+        _ = verify_custom_catalog_ingestion()
+
+        # Verify Models in Custom Source
+        for model_name in TEST_DATA["models"]:
+            try:
+                model_response = execute_get_command(
+                    url=f"{model_catalog_rest_url[0]}sources/{TEST_DATA['catalog_id']}/models/{model_name}",
+                    headers=model_registry_rest_headers,
+                )
+                # Successfully retrieving the model via the source-specific API endpoint
+                assert model_response["name"] == model_name, (
+                    f"Model name mismatch: expected {model_name}, got {model_response.get('name')}"
+                )
+            except ResourceNotFoundError as e:
+                pytest.fail(f"Model {model_name} not found in custom catalog {TEST_DATA['catalog_id']}: {e}")
+
+        # Verify Reconciliation Integrity - both catalogs should be present
+        current_configmap_data = ConfigMap(
+            name=DEFAULT_MODEL_CATALOG, client=admin_client, namespace=model_registry_namespace
+        ).instance.data
+
+        # Parse the current sources.yaml to verify structure
+        current_catalogs = yaml.safe_load(current_configmap_data["sources.yaml"])["catalogs"]
+        catalog_ids = [catalog["id"] for catalog in current_catalogs]
+
+        # Assert both default and custom catalogs are present
+        assert DEFAULT_CATALOG_ID in catalog_ids, f"Default catalog should be preserved. Found catalogs: {catalog_ids}"
+        assert TEST_DATA["catalog_id"] in catalog_ids, f"Custom catalog should be added. Found catalogs: {catalog_ids}"
+        assert len(catalog_ids) >= 2, f"Should have at least default and custom catalogs. Found: {catalog_ids}"
+
+        LOGGER.info("Custom catalog creation and verification completed successfully (permanent modification)")
```

Comment:
This would be problematic. We have custom and default in separate ConfigMap now.